theme/layouts/_sidebar.ejs

```diff
@@ -0,0 +1,35 @@
+<%
+const query = site.queryPages;
+subPages = site.getSubPages;
+
+const navigation = [];
+for (const menuItem of site.menuItems) {
+  const page = query(({originalPathname}) => originalPathname === menuItem)[0];
+  navigation.push(subPages(page));
+}
+
+const [currentPage] = query(({pathname}) => page.pathname === pathname);
+let currentPathname = currentPage.originalPathname.split("/");
+pathnameList = [];
+for (let i=currentPathname.length-1; i >= 0; i--) {
+  pathnameList.push(currentPathname.join("/"));
+  currentPathname = currentPathname.slice(0,i);
+}
+
+  function recNav(navig) { %>
+    <ul>
+      <% for (const {title, pathname, originalPathname, sub_items} of navig) { %>
+        <li <%- originalPathname.split("/").length == 1 ? "class='primary-menu'" : "" %>>
+          <a href="/<%= pathname %>"><%= title %></a>
+            <%
+            for (let i=0; i < pathnameList.length; i++) {
+              if (sub_items && originalPathname == pathnameList[i]) {
+                recNav(sub_items);
+                break;
+              }
+            } %>
```

Comment:
Not sure if this logic is required, here we need to ensure that item has subItems and those subItems are not children of current page.
```
if (sub_items && originalPathname !== currentOriginalPath) {
  recNav(sub_items);
}
```

Reply:
I think this does not replicate the Drupal website working logic. Not only `currentOriginalPath` matters for viewing children but also we need to specify the logic for "current" parent to view it's sections.